cmd/juju/cloud/add_test.go

```diff
@@ -7,172 +7,256 @@ import (
 	"bytes"
 	"io"
 	"io/ioutil"
-	"path/filepath"
 	"strings"
 
 	"github.com/juju/cmd"
+	jujutesting "github.com/juju/testing"
 	jc "github.com/juju/testing/checkers"
 	"github.com/pkg/errors"
 	gc "gopkg.in/check.v1"
 
+	cloudfile "github.com/juju/juju/cloud"
 	"github.com/juju/juju/cmd/juju/cloud"
-	"github.com/juju/juju/juju/osenv"
 	"github.com/juju/juju/testing"
 )
 
 type addSuite struct {
-	testing.FakeJujuXDGDataHomeSuite
+	jujutesting.IsolationSuite
 }
 
 var _ = gc.Suite(&addSuite{})
 
+func newFakeCloudQueryer() *fakeCloudQueryer {
+	return &fakeCloudQueryer{CallMocker: jujutesting.NewCallMocker()}
+}
+
+type fakeCloudQueryer struct {
+	cloud.CloudQueryer
+	*jujutesting.CallMocker
+}
+
+func (f *fakeCloudQueryer) ParseCloudMetadataFile(path string) (map[string]cloudfile.Cloud, error) {
+	results := f.MethodCall(f, "ParseCloudMetadataFile", path)
+	return results[0].(map[string]cloudfile.Cloud), typeAssertError(results[1])
+}
+
+func (f *fakeCloudQueryer) PublicCloudMetadata(searchPaths ...string) (result map[string]cloudfile.Cloud, fallbackUsed bool, _ error) {
+	results := f.MethodCall(f, "PublicCloudMetadata", searchPaths)
```

Comment:
Sorry, can you clarify a bit? What is the alternative to discovering a breakage by running the tests?

Also keep in mind that the mocking is for arguments and results, not method signatures. For that, we're still using a strongly-typed mock struct that satisfies the command's interface.